schema/query.go

```diff
@@ -356,3 +374,9 @@ func (e LowerOrEqual) Match(payload map[string]interface{}) bool {
 	n, ok := isNumber(getField(payload, e.Field))
 	return ok && (n <= e.Value)
 }
+
+// Match implements Expression interface
+func (e Regex) Match(payload map[string]interface{}) bool {
+	var regexpedValue = regexp.MustCompile(e.Value)
```

Comment:
You don't want to compile the regexp at runtime. The `Regex` type should store the compiled version of the regexp. This would also provide init time validation of the regexp.